datalad/core/distributed/clone.py

```diff
@@ -306,18 +306,48 @@ def __call__(
         if ds is not None:
             # we created a dataset in another dataset
             # -> make submodule
+            save_results = []
             for r in ds.save(
                     path,
+                    # Note, that here we know we don't save anything but a new
+                    # subdataset. Hence, don't go with default commit message,
+                    # but be more specific.
+                    message="[DATALAD] Added subdataset",
                     return_type='generator',
                     result_filter=None,
                     result_xfm=None,
                     on_failure='ignore'):
+                save_results.append(r)
                 yield r
 
             # Modify .gitmodules to contain originally given url. This is
-            # particularly relevant for postclone routines on a later `get` for
-            # that subdataset. See gh-5256.
-            ds.subdatasets(path, set_property=[("datalad-url", source)])
+            # particularly relevant for postclone routines on a later `get`
+            # for that subdataset. See gh-5256.
+            if any(r['action'] == 'save' and r['type'] == 'dataset' and \
+                   r['refds'] == ds.path for r in save_results):
```

Comment:
It seems that this is the only place where `save_results` is used. I am not sure how many it could be, but it seems a simpler approach would be to build the boolean indicator that triggers this conditional in the loop above. So something like

```
actually_saved_subds = None
...
actually_saved_subds = actually_saved_subds or r['status'] if \
    r['action'] == 'save' and r['type'] == 'dataset' and r['refds'] == ds.path else None
...
if actually_saved_subds:
    ...
```

This seems leaner and more clear as to why and how information is retained.

Reply:
Agree.

Reply:
Done. With slight variation.